group-coordinator/src/main/java/org/apache/kafka/coordinator/group/taskassignor/ProcessSpec.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.taskassignor;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ProcessSpec {
```

Comment:
I would not call this `ProcessSpec`. By "Spec" we mean a complete definition of the current state of the group, which is the input for the assignor. This isn't really part of the input to the assignor, but rather a helper class. 

Will we reuse this with the HighAvailablityTaskAssignor? Otherwise, we might just inline this class in the `StickTaskAssignor`, and just call it Process, wdyt?